src/parsec.rs

```diff
@@ -54,6 +57,7 @@ impl ParsecMap {
                 log_ident, gen_pfx_info
             );
         }
+        self.remove_old();
```

Comment:
consistent level of abstraction, either do not have a sub function `remove_old`, or add one for `add_new` (probably a better option).
```
fn init(... ) {
    self.add_new(...);
    self.remove_old();
}
```

Or probably better because otherwise the name is not quite right.
```
fn init(... ) {
    self.add_new(...);
    if self.need_remove_old() {
        self.remove_oldest();
    }
}

fn need_remove_old() {
    self.map.len() > MAX_PARSECS
}
```

Reply:
Yes good point, I agree!